src/org/sosy_lab/java_smt/solvers/z3/Z3FormulaCreator.java

```diff
@@ -386,16 +387,53 @@ public Long getFloatingPointType(FormulaType.FloatingPointType type) {
     return fpSort;
   }
 
+  private static class Z3AstReference extends PhantomReference<Z3Formula> {
+    private long z3Ast;
+    private @Nullable Z3AstReference prev;
+    private @Nullable Z3AstReference next;
+
+    // To generate dummy head and tail nodes
+    public Z3AstReference() {
```

Comment:
`public` here and for the methods makes no sense because the whole class is `final`.

Reply:
Changed everything to private.